backend/kernelspace/proxier.go

```diff
@@ -350,10 +349,9 @@ func NewProxier(
 	var sourceVip string
 	var hostMac string
 	if isOverlay(hnsNetworkInfo) {
-		// todo(knabben) - fix feature gates
-		/*if !true /*utilfeature.DefaultFeatureGate.Enabled(kubefeatures.WinOverlay) {
-			return nil, fmt.Errorf("WinOverlay feature gate not enabled")
-		}*/
+		if !config.EnableOverlay {
```

Comment:
I think overlay networking is supported on all Windows server versions now so we can probably drop all these checks too.

I think these checks were added because overlay networking was not originally available on Windows Server 2019 but was added after it was released with one of the SAC releases.

Reply:
In my previous comment i didn't realize we did a check on the HNS network type.
IMO that should be sufficient to distinguish between overlay networking and l2bridge